### DISCUSSION TITLE
Structural fundamentals of being a 'website'

### DISCUSSION THREAD

**Discussion by gbinal** [GSA-TTS/federal-website-standards#310] [Draft]
For any `x.gov` that intends to be a public-facing website, set a small number of clear, consistent structural expectations: 
- HTTPS required 
   - Problem this addresses: fundamental security need, and formal requirement all sites have
- Ensure that the base domain works both with or without `www.`  In other words, both https://www.x.gov and https://x.gov resolve and one redirects to the other.  
  - Problems this addresses: www. being required; the absence of www. being required; the two URLs resolving to different files.  
- For subdomains, if `www.` is used it cannot be required.  In other words, if https://www.something.agency.gov exists, the www. should not be required (https://something.agency.gov should work for a user).  
  - www. being required 
- A website should have a live homepage at its root.  In other words, if there is live content anywhere on a website (e.g. at https://program.gov/something/somethingelse), then the site should have a homepage (e.g. at https://program.gov).  
  - Confusion for a user seeking a website's main page; difficulty caused for scanning efforts.

**Comment by rphillippi** [comment]
Hi, we had talked about this before we did our initial sprint. I've added it to our [backlog](https://github.com/GSA-TTS/federal-website-standards/issues/311).

**Comment by thompsondc** [comment]
For the second bullet (facilitating the naked or www domain) - I suggest including a standard method of this; e.g.; CNAME record pointing from www to the naked domain.  I'm also not sure how/if the HSTS-strict policy would come into play, but the HTTP-requirement is a part of that:
https://digital.gov/2017/04/12/dotgov-domain-registration-program-to-provide-https-preloading-in-may/
